x-pack/plugins/lists/server/routes/delete_list_route.ts

```diff
@@ -28,7 +37,69 @@ export const deleteListRoute = (router: IRouter): void => {
       const siemResponse = buildSiemResponse(response);
       try {
         const lists = getListClient(context);
-        const { id } = request.query;
+        const exceptionLists = getExceptionListClient(context);
+        const { id, deleteReferences, ignoreReferences } = request.query;
+        let deleteExceptionItemResponses;
+
+        // ignoreReferences=true maintains pre-7.11 behavior of deleting value list without performing any additional checks
+        if (!ignoreReferences) {
+          const referencedExceptionListItems = await exceptionLists.findValueListExceptionListItems(
+            {
+              page: 1,
+              perPage: 10000,
+              sortField: undefined,
+              sortOrder: undefined,
+              valueListId: id,
+            }
+          );
+
+          if (referencedExceptionListItems?.data?.length) {
+            // Fetch referenced  exception lists so we
+            const referencedExceptionLists = await getReferencedExceptionLists(
```

Comment:
This isn't actually used outside of the `else` clause, so maybe move it down there so that it doesn't impact performance when it's not needed?

Reply:
Nice catch -- thank you! 🙂 Definitely unnecessary overhead if only going through the deleting references path.